src/js/components/stock-movement-wizard/AddItemsPage.jsx

```diff
@@ -401,6 +424,11 @@ class AddItemsPage extends Component {
         oldItem.recipient.id : oldItem.recipient;
       const newRecipient = item.recipient && _.isObject(item.recipient) ?
         item.recipient.id : item.recipient;
+      // Check if there is newly created item to add with the same product.id
+      const duplicatedItemIndex = _.findIndex(
+        lineItemsToBeAdded,
+        duplicate => _.get(duplicate, 'product.id') === _.get(item, 'product.id'),
```

Comment:
Seems like we need to be careful here. In the outbound case I think duplicates can be determined by product.id. However, I think it should probably be done for all three fields. And for inbound you could have the same product with a different inventory item (lot expiry), pack levels, and recipient. I also think we probably need to make a change to the server side to prevent duplicates and/or consolidate quietly.

Reply:
Yes, you're right. Should I remove those changes on frontend and move it to server side?